File: python/rosalind_hamm.py
from typing import Tuple

def read_data(filepath: str) -> Tuple[str, str]:
    with open(filepath, mode='r') as file:
        return tuple(file.read().split())

def write_data(filepath: str, data: int) -> None:
    with open(filepath, mode='w') as file:
        file.write(str(data))

def count_mutations(first_dna: str, second_dna: str) -> int:
    mutations = 0
    for index, symbol in enumerate(first_dna):
        if symbol != second_dna[index]:
            mutations += 1
    return mutations
    
if __name__ == '__main__':
    data = read_data('input/rosalind_hamm.txt')
    result = count_mutations(*data)
    write_data('output/rosalind_hamm.txt', result)

File: java/RosalindHAMM.java
import java.io.IOException;
import java.nio.file.*;
import java.util.List;

public class RosalindHAMM {
    private static List<String> readData(String filePath) throws IOException {
        Path inputFile = Paths.get(filePath);
        return Files.readAllLines(inputFile);
    }

    private static void writeData(String filePath, Integer data) throws IOException {
        Path outputFile = Paths.get(filePath);

        Files.deleteIfExists(outputFile);
        Files.createDirectories(outputFile.getParent());
        Files.createFile(outputFile);

        Files.writeString(outputFile, Integer.toString(data));
    }

    private static Integer countMutations(String firstDNA, String secondDNA) {
        Integer mutations = 0;
        for (int i = 0; i < firstDNA.length(); i++) {
            if (firstDNA.charAt(i) != secondDNA.charAt(i))
            mutations++;
        }
        return mutations;
    }

    public static void main(String[] args) throws IOException {
        List<String> data = readData("input/rosalind_hamm.txt");
        Integer result = countMutations(data.get(0), data.get(1));
        writeData("output/rosalind_hamm.txt", result);
    }
}
